### PR TITLE
docs: lift the decimal type under the number type

### PR DESCRIPTION
**Pull request by BohuTANG**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

* Add more examples for decimal data type, including min and max
* Lift the decimal order

Closes #issue

### CODE REVIEW

File: docs/doc/13-sql-reference/10-data-types/index.md [hunk 1]
```diff
@@ -7,36 +7,37 @@ slug: ./
...
 Databend supports three Semi-structured types: ARRAY, OBJECT and VARIANT.
 
-| Name    |    Aliases   | Build From Values       | Description
-|---------|--------------|-------------------------|----------------
-| ARRAY   |              | [1,2,3]                 | Zero-based indexed list, each value can have difference data type.
-| TUPLE   |              | ('2023-02-14 08:00:00','Valentine's Day') | Collection of ordered,immmutable, which requires the type of each element to be declared before being used.
-| VARIANT |  JSON        | [1,{"a":1,"b":{"c":2}}] | Collection of elements of different data types., including ARRAY and OBJECT.
+| Name    | Aliases | Build From Values                         | Description                                                                                                 |
+|---------|---------|-------------------------------------------|-------------------------------------------------------------------------------------------------------------|
+| ARRAY   |         | [1,2,3]                                   | Zero-based indexed list, each value can have difference data type.                                          |
```
Comment (sundy-li) [inline docs/doc/13-sql-reference/10-data-types/index.md:40]: VARIANT is the only SEMI struct datatype. `OBJECT` should be removed in the docs, let's use `VARIANT` instead.

Comment (sundy-li) [reply docs/doc/13-sql-reference/10-data-types/index.md:40]:
```
Zero-based indexed list, each value can have difference data type.
```

It's wrong,  we have `Array<T>` . It's inner type is fixed.

Comment (BohuTANG) [reply docs/doc/13-sql-reference/10-data-types/index.md:40]: Ok, let me remove it.

Comment (sundy-li) [reply docs/doc/13-sql-reference/10-data-types/index.md:40]: Also `Tuple` is not semi struct datatype, it's a nested datatype.

**Pull request merged**